Editor/EditorCore/ShapeComponentEditor.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEditor.UIElements;
+using UnityEngine;
+using UnityEngine.ProBuilder;
+using UnityEngine.UIElements;
+
+namespace UnityEditor.ProBuilder
+{
+    [CustomEditor(typeof(ShapeComponent))]
+    public class ShapeComponentEditor : Editor
+    {
+        private ShapeComponent m_ShapeComponent;
+        private IMGUIContainer m_ShapeField;
+
+        SerializedProperty m_shape;
+        static string[] s_ShapeTypes;
+        static TypeCache.TypeCollection s_AvailableShapeTypes;
+
+        static Pref<int> s_ActiveShapeIndex = new Pref<int>("ShapeBuilder.ActiveShapeIndex", 0);
+
+        static ShapeComponentEditor()
+        {
+            s_AvailableShapeTypes = TypeCache.GetTypesDerivedFrom<Shape>();
+            s_ShapeTypes = s_AvailableShapeTypes.Select(x => x.ToString()).ToArray();
```

Comment:
Assuming this is intended to be a menu item, should probably pass these through `ObjectNames.NicifyVariableName`